app/internal/corona_stats.py

```diff
@@ -0,0 +1,155 @@
+import json
+import random
+from datetime import date, datetime
+from typing import Any, Dict
+
+import httpx
+from fastapi import Depends
+from loguru import logger
+from sqlalchemy import desc, func
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from sqlalchemy.orm.exc import NoResultFound
+from sqlalchemy.sql.sqltypes import JSON
+
+from app.database.models import CoronaStats
+from app.dependencies import get_db
+
+DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+CORONA_API_URL = (
+    "https://datadashboardapi.health.gov.il/api/queries/vaccinated"
+)
+user_agent_list = [
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5)"
+    " AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.1.1 Safari/605.1.15",
+    "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:77.0)"
+    " Gecko/20100101 Firefox/77.0",
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) "
+    "AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537.36",
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:77.0) "
+    "Gecko/20100101 Firefox/77.0",
+    "Mozilla/5.0 (Windows NT 10.0; Win64; x64) "
+    "AppleWebKit/537.36 (KHTML, like Gecko)"
+    " Chrome/83.0.4103.97 Safari/537.36",
+]
```

Comment:
Why do we need this? From quick trial the API plays nice with Python

Reply:
At some point I started to get 403 from the code while it worked fine from the browser, I get it to work again only with different user agents

Reply:
You use only a single user agent from the list. Maybe move the `headers` to be inside of the functions

Reply:
oh! you are right! i misse that.